### DISCUSSION TITLE
Vercel Deployment Logs (Real-time)

### DISCUSSION THREAD

**Discussion by RutsuKun** [aarondill/Vercel-Project-Manager#11] [Ideas]
Possibility to observe deployments logs in Visual Studio Code, it's nice if they were Real-time.

**Comment by aarondill** [comment]
it would likely be a very complicated process to sync the logs in real-time (would we have to register a log drain?)

alternatively, perhaps we could open a terminal with `vercel logs --follow URL`

if you decide to implement this, i'd be happy to merge it, otherwise, I think i'll leave this feature to the Vercel website.

**Reply by RutsuKun** [reply]
I have no experience in creating extensions for Visual Studio Code, but I'll try to come up with something.

**Reply by aarondill** [reply]
the terminal approach would be fairly easy, look at `src/commands/cli/VercelDev.ts`, `src/commands/cli/VercelLink.ts`, and `src/commands/cli/VercelDeploy.ts`. The harder part would be getting a URL to pass to it (get input from the user?)

Or, if you wanted to create a button to open it then look at `src/commands/CopyURL.ts` and add it to package.json:
```json
      "view/item/context": [
        {
          "command": "vercel.copyURL",
          "when": "view == vercel-deployments && viewItem == deployment || viewItem == deploymentLog",
          "group": "inline@2"
        },
        {
          "command": "vercel.removeEnvironment",
          "when": "view == vercel-environment && viewItem == Environment && vercelLinked",
          "group": "inline@1"
        },
        {
          YOUR COMMAND HERE
       }
      ]
```

to actually display them in a web panel would be a much harder process, which I can't necessarily help with, since I actually don't know that part of Vercel's API.
